kustomize/commands/edit/add/configmap.go

```diff
@@ -142,6 +143,10 @@ func addConfigMap(
 	// Validate by trying to create corev1.configmap.
 	args.Options = types.MergeGlobalOptionsIntoLocal(
 		args.Options, k.GeneratorOptions)
+
+	// guarantee end order is predictable
+	slices.Sort(args.LiteralSources)
```

Comment:
Do you mind explaining the motivation here? It seems to me that preserving the order of the user's original input might be more desirable than sorting it

Reply:
This is mostly for testing purposes 😅 I can roll it back.

Reply:
This was rolled back.